packages/cli/src/lib/log.ts

```diff
@@ -122,6 +167,122 @@ export function muteAuthLogs({
   };
 }
 
+/**
+ * Modify logs from Hydrogen to use cli-kit banners
+ * Format: `[h2:type:scope] message`. Example: `[h2:error:storefront.query] Wrong query`
+ * Where the message can be multiline and the last line
+ * can contain links to docs or other resources.
+ */
+export function enhanceH2Logs(options: {
```

Comment:
Is there a reason to rely on a special prefix to logged messages, instead of just export and importing a special function to log what we need, like a `logGraphQLError()` or something like that?

Reply:
Basically we don't want to bundle dev-only stuff in H2 or HR. This way, we only need to add a simple prefix that can be even readable in production or in dev without H2 CLI. If you are in dev withour CLI, then you get them replaced.
It's basically not intrusive, I think.

Reply:
Then in production, you get an error that includes that prefix?

![image](https://github.com/Shopify/hydrogen/assets/1566869/2da42f63-fc7b-4220-9d30-e1ccd2a63e80)

Reply:
Yes, that's the idea. Hopefully we can use that in Oxygen as well.
Do you think we shouldn't add them in production? We can do that as well but I think it's actually useful 🤔

Reply:
No, I think it will be useful in Oxygen. 🚢